src/analyzing/AST_to_IL.ml

```diff
@@ -1027,16 +1027,32 @@ and record env ((_tok, origfields, _) as record_def) =
                    ( { G.name = G.EN (G.Id (id, _)); tparams = None; _ },
                      def_kind );
                _;
-             } ->
-             let fdeforig =
+             } as forig ->
+             let field_def =
                match def_kind with
                (* TODO: Consider what to do with vtype. *)
                | G.VarDef { G.vinit = Some fdeforig; _ }
                | G.FieldDefColon { G.vinit = Some fdeforig; _ } ->
-                   fdeforig
+                   expr env fdeforig
+               | G.FuncDef fdef ->
+                   (* TODO: we should have a use def.f_tok *)
+                   let tok = G.fake "record_literal_funcdef" in
+                   let lval = fresh_lval env tok in
+                   let fdef =
+                     (* This is a recursive call to `function_definition` and we need to pass
```

Comment:
There is another such comment attached to `| G.Lambda fdef ->`. It would be better if we had the comment once, with some kind of tag, and then just refer to this. For example, in `Nosemgrep.ml` me have `NOTE(multiple)` and in some other place we refer to it as `See NOTE(multiple)`.